orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -551,9 +552,18 @@ Status OrtLoadInternal(const PathString& checkpoint_path,
     LoadTensorProtoFromFile(tensor_file_full_path, tensor_protos, "[params]");
 
     for (const auto& tensor_proto : tensor_protos) {
-      param_tensor_protos.push_back(tensor_proto);
+      param_tensor_protos.insert(std::make_pair(tensor_proto.name(), tensor_proto));
     }
   }
+
+  // Load imported initializers into the Model
+  for (auto& init : *(model_proto.mutable_graph()->mutable_initializer())) {
+    if (!init.has_name()) ORT_THROW("Initializer name is missing.");
+    if (!param_tensor_protos.count(init.name())) ORT_THROW("Invalid initializer name.");
+    auto it = param_tensor_protos.find(init.name());
```

Comment:
Do you need this find call given on line 562 you are checking initializer exists?

Reply:
This check is to make sure that the checkpoint params names are matching the model's params names.

Reply:
You dont need both 563 and 564. You can skip 563 and simply do a find and check it != param_tensor_protos.end() to make sure there exists a param with name == init.name